File: Java/q1117.java
// Validação de Nota

import java.util.Scanner;

public class Main {
 
    public static void main(String[] args) {
 
         Scanner input = new Scanner(System.in);
         int notasValidas = 0;
         double media = 0;
         
         do {
             double nota = input.nextDouble();
             
             if(nota >= 0 && nota <= 10){
                 notasValidas++;
                 media += nota;
             } else {
                 System.out.println("nota invalida");
             }
         } while(notasValidas < 2);
         
         media = media/2.0;
         
         System.out.printf("media = %.2f%n", media);
    }
}

File: Cpp/q1117.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main(){
  int v = 0;
  float nota, n=0, media;
  
  while(true){
    if(v == 2){ break;}
    cin >> nota;
    if(v == 0){
        if(nota >= 0 && nota <= 10){
            n += nota;
            v++;
        } else {
            cout << "nota invalida" << endl;
        }
    } else {
        if(nota >= 0 && nota <= 10){
            n += nota;
            media = (n/2.0);
            v++;
        } else {
            cout << "nota invalida" << endl;
        }
    }
}
  cout << fixed << setprecision(2);
  cout << "media = " << media << endl;
  return 0;
}